.github/workflows/overlay.toml

```
@@ -115,23 +118,48 @@ url = "https://typora.io/#linu://typora.io/releases/stable.html"
 regex = "typora_([\\d.]+)_amd64.deb"
 github_account = "CHN-beta"
 
-# TODO:
-#["app-emulation/deepin-udis86"]
-
-# TODO:
-#["app-emulation/deepin-wine-helper"]
-
-# TODO:
-#["app-emulation/deepin-wine-plugin"]
-
-# TODO:
-#["app-emulation/deepin-wine5"]
-
-# TODO:
-#["app-emulation/deepin-wine6-stable"]
-
-# TODO: PV mismatching tags
-#["app-emulation/looking-glass"]
+["app-emulation/deepin-udis86"]
+source = "regex"
+url = "https://community-packages.deepin.com/deepin/pool/non-free/u/udis86"
+regex = '''udis86[-_\.]([\d\w]+(?:[\._-][\d\w]+)*)(?=\.tar\..+|\.zip|\.tgz|\.deb|\.AppImage)'''
+from_pattern = '''(.*)-(.*)_(.*)'''
+to_pattern = '''\1_p\2'''
```

```suggestion
["app-emulation/deepin-udis86"]
source = "apt"
mirror = "https://community-packages.deepin.com/deepin/"
suite = "apricot"
repo = "non-free"
pkg = "udis86"
from_pattern = '''(.*)-(.*)'''
to_pattern = '''\1_p\2'''
```

可以用 source = "apt" 直接读取 deepin apt 仓库，不需要自己写正则。

---

星火商店不知道是否也可以用 sources = "apt"，可能不行。

---

这种冷门的apt, jq, debian, 甚至gitlab, 都不如regex或者cmd 方便.

---

> 这种冷门的apt, jq, debian, 甚至gitlab, 都不如regex或者cmd 方便, 没必要改. regex和cmd反而比较清晰.
不需要写成apt, 反而用正则, 和源码一致, 多一层封装而已.

不要自己封装，尽量使用 nvchecker 内置的选项。

> from_pattern = '''(.)-(.)'''
> to_pattern = '''\1_p\2'''
> 你的这个from_pattern 写错的.

正确的是？

["app-emulation/deepin-udis86"]
source = "apt"
mirror = "https://community-packages.deepin.com/deepin/"
suite = "apricot"
repo = "non-free"
pkg = "udis86"
from_pattern = '''(.*)-(.*)_(.*)'''
to_pattern = '''\1_p\2'''

[I 06-08 16:05:30.445 core:416] app-emulation/deepin-udis86: updated to 1.72-4


["app-emulation/deepin-udis86"]
source = "apt"
mirror = "https://community-packages.deepin.com/deepin/"
suite = "apricot"
repo = "non-free"
pkg = "udis86"
from_pattern = '''(.*)-(.*)'''
to_pattern = '''\1_p\2'''

[I 06-08 16:03:35.766 core:416] app-emulation/deepin-udis86: updated to 1.72_p4

---

正确是的要把 `'''(.*)-(.*)_(.*)'''` 第三个group去掉的.

> [udis86_1.72-4_amd64.deb](https://cdn-community-packages.deepin.com/deepin/pool/non-free/u/udis86/udis86_1.72-4_amd64.deb)                            11-Apr-2022 12:06     796
> [udis86_1.72-4_i386.deb](https://cdn-community-packages.deepin.com/deepin/pool/non-free/u/udis86/udis86_1.72-4_i386.deb)                             11-Apr-2022 12:05     34K

---

source = "apt" 会过滤架构（默认是 amd64），所以不再需要 `_(.*)`

---

好吧, 我就用apt把, 看起来应该没有bug